tests/rptest/tests/enterprise_features_license_test.py

```diff
@@ -202,10 +179,13 @@ def has_role(r: str):
                     "openssl_module_directory":
                     self.redpanda.get_openssl_modules_directory()
                 })
-        elif feature == Features.datalake_iceberg:
+        elif feature == Feature.datalake_iceberg:
             self.redpanda.set_cluster_config({'iceberg_enabled': 'true'},
                                              expect_restart=True)
-        elif feature == Features.leadership_pinning:
+        elif feature == Feature.leadership_pinning:
+            self.redpanda.set_cluster_config(
+                {"default_leaders_preference": "racks:rack1"},
+                expect_restart=True)
```

Comment:
This is needed because creating a topic with `"redpanda.leaders.preference"` won't error for `test_enable_features` as expected, right? Not for this PR, but I think we should error on the `create_topic` request below as well.

Reply:
yup. i think that's what @IoannisRP is working on if I heard right in standup. the nice thing about all this junk is that everything we change causes a test to fail, so you know immediately where the DT changes ought to go 😁

Reply:
there are 2 different things here. The `"redpanda.leaders.preference"` topic preference and the general config property `"default_leaders_preference"`. I am working on disabling the later. It doesn't affect topic creation. 

We do have a ticket about preventing topic creations under sanctions, though.